docs/configuration/tools/router_check.rst

```diff
@@ -0,0 +1,109 @@
+.. _config_tools_router_check_tool:
+
+Route table check tool
+======================
+
+**NOTE: The following configuration is for the router table check tool only and is not part of the Envoy binary.
+The route table check tool is used for testing purposes only.**
+
+Input for the route table check tool. The route table check tool checks if the route returned
+by a router matches what is expected. The tool can be used to check cluster name, virtual cluster name,
+virtual host name, path rewrite, host rewrite, and path redirect matches. Extensions for other
+test cases can be added. The "check" field specifies the expected values in each test case. At least one test
+case is required. In addition, the authority and path fields specify the url sent to the router
+and are also required. A simple configuration has one test case and is writen as follows. The test
+expects a cluster name match of "instant-server".::
+
+   [
+     {
+       "authority":"api.lyft.com",
+       "path": "/api/locations",
+       "check": {"cluster_name": "instant-server"}
+     }
+   ]
+
+.. code-block:: json
+
+  [
+    {
```

Comment:
I was doing a bit of thinking about this, and maybe the print output could be changed to something that looks like 
```
Test Case: <test case name>
P/F <test_case> <expected> <actual>
...
```
The one thing that I'm still not sure about is if we should print the test cases that passed.